sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3014,6 +3014,22 @@ object GpuOverrides extends Logging {
 
         override val supportBufferConversion: Boolean = true
       }),
+    expr[StddevPop](
+      "Aggregation computing population standard deviation",
+      ExprChecks.aggNotReduction(
+        TypeSig.DOUBLE, TypeSig.DOUBLE,
+        Seq(ParamCheck("input", TypeSig.gpuNumeric, TypeSig.numeric))),
```

Comment:
Technically the only value supported by StddevPop as input in spark is Double

https://github.com/apache/spark/blob/adbea252db0f61ed3aaa45ea20b704bad6c47408/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/CentralMomentAgg.scala#L58

Could you please update the type signature here and on StddevSamp to only show DOUBLE as the supported input type by both us and Spark?  This should also make testing a lot simpler because you don't need to worry about all of the different types.  Spark will cast them ahead of time to a Double for us.

Reply:
Hah I thought that if `TypeSig` is not double then Spark will throw an exception instead if the input is not double (such as int). Thanks for letting me know this.

Reply:
There are different classes in Spark that indicate if type coercion is allowed or not. `ImplicitCastInputTypes` tells Spark that it can cast the inputs to try and match the desired input. There are also several rules about what types can automatically be cast and which cannot `TypeCoersion.scala`.